core/corehttp/gateway_handler.go

```diff
@@ -417,6 +410,7 @@ func (i *gatewayHandler) getOrHeadHandler(w http.ResponseWriter, r *http.Request
 		}
 	}
 
+	// TODO(JJ): Should this be somewhere else for UnixFS, given the redirects handling above?
```

Comment:
I think it is fine keeping it here – will measure time to first logical block, no matter if any additional logic like `_redirects` was applied or not.